scripts/test/Muon/MaxEntPresenter_test.py

```diff
@@ -7,140 +7,112 @@
 import unittest
 
 from unittest import mock
-from Muon.GUI.Common.utilities import load_utils
 from Muon.GUI.Common import thread_model
-from Muon.GUI.FrequencyDomainAnalysis.MaxEnt import maxent_presenter
-from Muon.GUI.FrequencyDomainAnalysis.MaxEnt import maxent_view
+from Muon.GUI.Common.test_helpers.context_setup import setup_context
+from Muon.GUI.FrequencyDomainAnalysis.MaxEnt import maxent_presenter_new
+from Muon.GUI.FrequencyDomainAnalysis.MaxEnt import maxent_view_new
 from Muon.GUI.FrequencyDomainAnalysis.MaxEnt import maxent_model
 
 
-def test(inputs):
-    inputs["OutputPhaseTable"] = "test"
-
-
 class MaxEntPresenterTest(unittest.TestCase):
     def setUp(self):
-        self.load=mock.create_autospec(load_utils.LoadUtils,spec_set=True)
-        self.load.getCurrentWS=mock.Mock(return_value=["TEST00000001",["fwd","bkwd"]])
-        self.load.hasDataChanged = mock.MagicMock(return_value=False)
-
-        self.model=mock.create_autospec(maxent_model.MaxEntModel,spec_set=True)
-
-        self.view=mock.create_autospec(maxent_view.MaxEntView,spec_set=True)
-        #signals
-        #needed for connect in presenter
-        self.view.maxEntButtonSignal=mock.Mock()
-        self.view.cancelSignal=mock.Mock()
-        self.view.phaseSignal=mock.Mock()
+        self.context = setup_context(True)
+        self.context.data_context.instrument = 'MUSR'
+        self.context.gui_context.update({'RebinType': 'None'})
+
+        # Model
+        self.model = mock.create_autospec(maxent_model.MaxEntWrapper, spec_set=True)
+        self.model.cancel = mock.Mock()
+
+        # View
+        self.view = mock.create_autospec(maxent_view_new.MaxEntView, spec_set=True)
+        # signals
+        # needed for connect in presenter
+        self.view.maxEntButtonSignal = mock.Mock()
```

Comment:
These lines aren't needed since the autospec will create and attach mock objects to them anyway (as they are class definitions) and it adds noise.

The only ones which do need these manually setting is anything created in `__init__` as we don't construct the object so mock cannot see those fields (such as `view.run` / `view.table` ..etc.)

Reply:
Ah okay thank you! I am still new to mocking so was following the same design pattern as before.